xwiki-platform-core/xwiki-platform-uiextension/xwiki-platform-uiextension-api/src/main/java/org/xwiki/uiextension/internal/WikiUIExtensionParameters.java

```diff
@@ -123,12 +127,18 @@ public WikiUIExtensionParameters(String id, String rawParameters, ComponentManag
      */
     private Map<String, String> parseParameters(String rawParameters)
     {
-        Map<String, String> result = new HashMap<String, String>();
-        for (String line : rawParameters.split("[\\r\\n]+")) {
-            String[] pair = line.split("=", 2);
-            if (pair.length == 2 && !"".equals(pair[0]) && !"".equals(pair[1])) {
-                result.put(pair[0], pair[1]);
+        Map<String, String> result;
+        Properties properties = new Properties();
+        try {
+            properties.load(new StringReader(rawParameters));
+            result = new HashMap<>();
+            for (String key : properties.stringPropertyNames()) {
+                if (!Objects.equals(key, "")) {
+                    result.put(key, properties.getProperty(key));
+                }
             }
+        } catch (IOException e) {
```

Comment:
We shouldn't ignore silently exception: at least add a logger to inform about it.

Reply:
Why convert to Map? Why not return a Properties object? In the code we iterate over it later on so you're doing 2 iterations instead of one from what I see.